arch/eventloop_posix_udp.c

```diff
@@ -445,6 +447,83 @@ UDP_shutdownConnection(UA_ConnectionManager *cm, uintptr_t connectionId) {
     return UA_STATUSCODE_GOOD;
 }
 
+/**
+ * Retrieves hostname and port from given key value parameters.
+ * @param[in] paramsSize the size of the parameter list
+ * @param[in] params the parameter list to retrieve from
+ * @param[out] hostname the retrieved hostname when present, NULL otherwise
+ * @param[out] portStr the retrieved port when present, NULL otherwise
+ * @param[in] logger the logger to log information
+ * @return -1 upon error, 0 if there was no host or port parameter, 1 if
+ *         host and port are present
+ */
+static int
+getHostAndPortFromParams(size_t paramsSize, const UA_KeyValuePair *params,
+                         char *hostname, char *portStr, const UA_Logger *logger) {
+    /* Prepare the port parameter as a string */
+    const UA_UInt16 *port = (const UA_UInt16*)
+        UA_KeyValueMap_getScalar(params, paramsSize,
+                                 UA_QUALIFIEDNAME(0, "port"),
+                                 &UA_TYPES[UA_TYPES_UINT16]);
+    if(!port) {
+        UA_LOG_DEBUG(logger, UA_LOGCATEGORY_NETWORK,
+                     "UDP\t| No port found");
+        return 0;
+    }
+    UA_snprintf(portStr, UA_MAXPORTSTR_LENGTH, "%d", *port);
+
+    /* Prepare the hostname string */
+    const UA_String *host = (const UA_String*)
+        UA_KeyValueMap_getScalar(params, paramsSize,
+                                 UA_QUALIFIEDNAME(0, "hostname"),
+                                 &UA_TYPES[UA_TYPES_STRING]);
+    if(!host) {
+        UA_LOG_DEBUG(logger, UA_LOGCATEGORY_NETWORK,
+                     "UDP\t| No hostname found");
+        return 0;
+    }
+    if(host->length >= UA_MAXHOSTNAME_LENGTH) {
+        UA_LOG_ERROR(logger, UA_LOGCATEGORY_EVENTLOOP,
+                     "UDP\t| Open UDP Connection: Hostname too long, aborting");
+        return -1;
+    }
+    strncpy(hostname, (const char*)host->data, host->length);
+    hostname[host->length] = 0;
+    return 1;
+}
+
+static int
+getConnectionInfoFromParams(size_t paramsSize, const UA_KeyValuePair *params, char *hostname, char *portStr,
+                            struct addrinfo **info, const UA_Logger *logger) {
+
+    int foundParams = getHostAndPortFromParams(paramsSize, params, hostname, portStr, logger);
+    if (foundParams < 0) {
+        UA_LOG_ERROR(logger, UA_LOGCATEGORY_EVENTLOOP,
+                     "UDP\t| Failed retrieving host and port parameters");
+        return -1;
+    }
+    if(foundParams == 0) {
+        UA_LOG_DEBUG(logger, UA_LOGCATEGORY_EVENTLOOP,
+                     "UDP\t| Hostname or Port not present");
+        return 0;
+    }
+    /* Create the socket description from the connectString
+    * TODO: Make this non-blocking */
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(struct addrinfo));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_DGRAM;
+    int error = getaddrinfo(hostname, portStr, &hints, &*info);
```

Comment:
This `&*info` is a strange construction.
Are you sure about it?
It would need a short comment so that future generations are not entirely confused.

Reply:
Good call, this was not intended to have happened :D